Added -Version parameter to Install-BicepCLI

StefanIvemo opened the pull request:
- Added -Version parameter to `Install-BicepCLI` to be able to specify which version to install.
- Parameter validation and argument completer implemented for `-Version` parmeter.
- Updated `Install-BicepCLI` help files
- Renamed private funtion `LatestBicepVersion` to `ListBicepVersions`. Function now returns a list of all available versions or the latest.
- Updated all functions depending on `LatestBicepVersion`  to use `ListBicepVersions`.

JohnRoos commented:
Should the version contain the initial "v"? I find it more natural to remove the "v". That would make it compatible with the version class as well if anyone would need it in the future (eg. [version]'0.2.328').

StefanIvemo commented:
> Should the version contain the initial "v"? I find it more natural to remove the "v". That would make it compatible with the version class as well if anyone would need it in the future (eg. [version]'0.2.328').

Good point! I added the v (which I originally removed) because I was inspired by how the Azure CLI bicep commands allowed specifying the version to install using that format.

StefanIvemo commented:
- Removed the v from version to allow the use of the `[version] ` class
- Removed the unused function `LatestBicepVersion` (replaced in this PR)
- Updated test `LatestBicepVersion.Tests.ps1`  to `ListBicepVersions.Tests.ps1` and fixed the tests to work with the new function
